pykrige/rk.py

```diff
@@ -444,3 +453,294 @@ def score(self, p, x, y, sample_weight=None, **kwargs):
+        )
+
+
+def closure(data, k=1.0):
+    """Apply closure to data, sample-wise. 
+    Adapted from https://github.com/ofgulban/compoda.
+
+    Parameters
+    ----------
+    data : 2d numpy array, shape [n_samples, n_measurements]
+        Data to be closed to a certain constant. Do not forget to deal with
+        zeros in the data before this operation.
+    k : float, positive
+        Sum of the measurements will be equal to this number.
+
+    Returns
+    -------
+    data : 2d numpy array, shape [n_samples, n_measurements]
+        Closed data.
+
+    Reference
+    ---------
+    [1] Pawlowsky-Glahn, V., Egozcue, J. J., & Tolosana-Delgado, R.
+        (2015). Modelling and Analysis of Compositional Data, pg. 9.
+        Chichester, UK: John Wiley & Sons, Ltd.
+        DOI: 10.1002/9781119003144
+    """
+    data_sum = np.sum(data, axis=1)
+    out = np.copy(data)
+    for i in range(data.shape[1]):
+        out[:, i] = np.divide(out[:, i], data_sum)
+    out = out * k
+    return out
+
+
+def ilr_transformation(data):
+    """Isometric logratio transformation (not vectorized).
+    Adapted from https://github.com/ofgulban/compoda.
+
+    Parameters
+    ----------
+    data : 2d numpy array, shape [n_samples, n_coordinates]
+        Barycentric coordinates (closed) in simplex space.
+
+    Returns
+    -------
+    out : 2d numpy array, shape [n_samples, n_coordinates-1]
+        Coordinates in real space.
+
+    Reference
+    ---------
+    [1] Pawlowsky-Glahn, V., Egozcue, J. J., & Tolosana-Delgado, R.
+        (2015). Modelling and Analysis of Compositional Data, pg. 37.
+        Chichester, UK: John Wiley & Sons, Ltd.
+        DOI: 10.1002/9781119003144
+    """
+    data = data + np.finfo(float).eps
+    dims = data.shape
+    out = np.zeros((dims[0], dims[1]-1))
+    helmertian = -helmert(dims[1]).T
+    for i in range(data.shape[0]):
+        out[i, :] = np.dot(np.log(data[i, :]), helmertian)
+    return out
+
+
+def inverse_ilr_transformation(data):
+    """Inverse isometric logratio transformation (not vectorized).
+    Adapted from https://github.com/ofgulban/compoda.
+
+    Parameters
+    ----------
+    data : 2d numpy array, shape [n_samples, n_coordinates]
+        Isometric log-ratio transformed coordinates in real space.
+
+    Returns
+    -------
+    out : 2d numpy array, shape [n_samples, n_coordinates+1]
+        Barycentric coordinates (closed) in simplex space.
+
+    Reference
+    ---------
+    [1] Pawlowsky-Glahn, V., Egozcue, J. J., & Tolosana-Delgado, R.
+        (2015). Modelling and Analysis of Compositional Data, pg. 37.
+        Chichester, UK: John Wiley & Sons, Ltd.
+        DOI: 10.1002/9781119003144
+    """
+    data = data + np.finfo(float).eps
+    dims = data.shape
+    out = np.zeros((dims[0], dims[1]+1))
+    helmertian = -helmert(dims[1]+1)
+    for i in range(data.shape[0]):
+        out[i, :] = np.exp(np.dot(data[i, :], helmertian))
+    return closure(out)
```

Comment:
You could just use `np.einsum` here to vectorize this operation:
```python
out = np.einsum("ij,jk->ik", data, helmertian)
```
Then you also don't need to initalize `out`.